test/e2e/provider_expectations.go

```diff
@@ -298,3 +306,316 @@ func (tc *testConfig) ExpectNodeToBeDrainedBeforeDeletingMachine() error {
 		return true, nil
 	})
 }
+
+var nodeDrainLabels = map[string]string{
+	workerRoleLabel:      "",
+	"node-draining-test": "",
+}
+
+func machineFromMachineset(machineset *mapiv1beta1.MachineSet) *mapiv1beta1.Machine {
+	randomUUID := string(uuid.NewUUID())
+
+	machine := &mapiv1beta1.Machine{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: machineset.Namespace,
+			Name:      "machine-" + randomUUID[:6],
+			Labels:    machineset.Labels,
+		},
+		Spec: machineset.Spec.Template.Spec,
+	}
+	if machine.Spec.ObjectMeta.Labels == nil {
+		machine.Spec.ObjectMeta.Labels = map[string]string{}
+	}
+	for key := range nodeDrainLabels {
+		if _, exists := machine.Spec.ObjectMeta.Labels[key]; exists {
+			continue
+		}
+		machine.Spec.ObjectMeta.Labels[key] = nodeDrainLabels[key]
+	}
+	return machine
+}
+
+func replicationControllerWorkload(namespace string) *corev1.ReplicationController {
+	var replicas int32 = 20
+	return &corev1.ReplicationController{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "pdb-workload",
+			Namespace: namespace,
+		},
+		Spec: corev1.ReplicationControllerSpec{
+			Replicas: &replicas,
+			Selector: map[string]string{
+				"app": "nginx",
+			},
+			Template: &corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "nginx",
+					Labels: map[string]string{
+						"app": "nginx",
+					},
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:    "work",
+							Image:   "busybox",
+							Command: []string{"sleep", "10h"},
+							Resources: corev1.ResourceRequirements{
+								Requests: corev1.ResourceList{
+									"cpu":    resource.MustParse("50m"),
+									"memory": resource.MustParse("50Mi"),
+								},
+							},
+						},
+					},
+					NodeSelector: nodeDrainLabels,
+					Tolerations: []corev1.Toleration{
+						{
+							Key:      "kubemark",
```

Comment:
kubemark?

Reply:
Yes, so the same test can by used by both aws and kubemark. The toleration has no effect as long as node is not tainted.